lgc/state/PipelineState.cpp

```diff
@@ -1121,7 +1121,28 @@ unsigned PipelineState::getShaderWaveSize(ShaderStage stage) {
   assert(stage <= ShaderStageCompute);
   if (!m_waveSize[stage])
     setShaderDefaultWaveSize(stage);
-  return m_waveSize[stage];
+
+  unsigned waveSize = m_waveSize[stage];
+
+  if (stage == ShaderStageVertex) {
+    if (hasShaderStage(ShaderStageTessControl)) {
+      waveSize = std::max(waveSize, m_waveSize[ShaderStageTessControl]);
+    } else if (hasShaderStage(ShaderStageGeometry)) {
+      waveSize = std::max(waveSize, m_waveSize[ShaderStageGeometry]);
+    }
+  } else if (stage == ShaderStageTessEval) {
+    if (hasShaderStage(ShaderStageGeometry)) {
+      waveSize = std::max(waveSize, m_waveSize[ShaderStageGeometry]);
+    }
+  } else if (stage == ShaderStageGeometry) {
+    waveSize = std::max(waveSize, m_waveSize[ShaderStageVertex]);
```

Comment:
Make this clear
} else if (stage == ShaderStageGeometry) {
  // NGG, no geometry
  if (!hasShaderStage(ShaderStageGeometry)
    waveSize = std::max(waveSize, m_waveSize[hasShaderStage(ShaderStageTessEval) ? ShaderStageTessEval : ShaderStageVertex];
}